doc/ocaml-ecosystem.rst

```diff
@@ -0,0 +1,89 @@
+
+The Compiler Distribution: Compiling and Linking
+------------------------------------------------
+
+The compiler distribution contains "core" tools including the compilers
+(``ocamlc`` and ``ocamlopt``). They turn source files (with extensions ``.ml``
+and ``.mli``) into executables and libraries. Dependencies between compiled
+objects only exist at the module level, so this is a low-level tool.
+
+Findlib: Metadata for libraries
+-------------------------------
+
+Findlib is a tool that defines the concept of library, so that libraries can
+depend on other libraries on top of the notion of module. Definitions of
+libraries, and other pieces of metadata, are stored in ``META`` files.
+
+Findlib ships an executable named ``ocamlfind`` that can be used as a wrapper
+on top of the compilers, to perform tasks such as producing an executable from
+compiled object files and external libraries.
+
+Opam: a Collection of Software Projects
+---------------------------------------
+
+Opam is a package manager. It is used to determine which packages are
+necessary, and how to fetch and build them. Packages can contain libraries,
+executables, and other kinds of files.
+
+The notion of version is specific to opam: if your project uses a function
+named ``Png.read_file`` but this function has been added only in version
+``1.2.0`` of that package, opam needs to know about it.
+
+OPAM manages collections of installed packages, called switches. Using your
+project's dependencies (names and version constraints), it is able to create a
+switch that you'll be using to develop your project.
+
+Public definition of packages are available in a database called
+opam-repository which is maintained as a public git repository. Publishing a
+package on opam (to make sure that external users can use your project)
+consists in adding its definition to opam-repository.
+
+Dune: Giving Structure to your Source Tree
+------------------------------------------
+
+Dune is a build system. It is used to orchestrate the compilation of source
+files into executables and libraries.
+
+Assuming you have a development switch set up, you communicate to Dune how your
+project is organized in terms of executables, libraries and tests. It is then able to assemble the source files of your projects with the dependencies installed in an opam switch, to create compiled assets for your project.
+
+How Dune Integrates with the Ecosystem
+--------------------------------------
+
+Dune is designed to integrate with this set of tools:
+
+- by knowing how the compilers operate, it knows which build commands should be
+  re-executed if some source files change.
+- it outputs metadata like dependency information into ``META`` files that
+  Findlib is able to use. This ensures that a project that does not use Dune
+  can use a library that has been produced by Dune. Conversely, it can read
+  these files to determine dependency information for dependencies that have
+  not been produced by Dune.
```

Comment:
```suggestion
- It outputs metadata like dependency information into ``META`` files that
  Findlib is able to use. This ensures that even if a project does not use Dune, it
  can use a library that has been produced by Dune. Conversely, it can read
  these files to determine dependency information for dependencies that have
  not been produced by Dune.
```